.vscode/README.md

```diff
@@ -0,0 +1,39 @@
+
+## Words and word boundaries
+
+"-" and "." are word boundaries. For example: An artifactId, azure-core-amqp, will verify azure, core and amqp.AMQP, not being in the dictionary would get flagged and that's the only word that would be required to add to the list.
+
+## Case insensitivity
+
+The ignored words are case insensitive. AMQP, amqp, Amqp etc would all be picked up from "amqp"
+
+## Regular expressions
+
+There are times when a single word really isn't enough. For example: Instead of adding each unique word from each and every groupId and artifactId from every Azure SDK library and external dependency, it's cleaner to regex away the line that contains the artifactId. Always match
+
+### Multi-line regular expression
+
+Multi-line regular expressions should be used with extreme caution as it's far too easy to accidentally regex away too much. Any multi-line regular expression should have hard tags that delineate a clear beginning and a clear ending with the multi-line regular expression in between these tags. These tags should be as close to what you want to regex away as possible. For example: A `<properties>` tag may contain a number other tags that you with to regex away but they can also contain things, like a `<legal>`, which contains a string that needs to get checked. For any set of tags that may contain sub-tags, only the sub-tags should be regex'd unless you can guarantee that the all of possible sub-tags will never need to be scanned. `<compilerArguments>` would be an example where all of the sub-tags would be skipped.
+
+**If a multi-line regular expression is needed then a member of the Azure SDK Engineering System Team should be added to the code review.**
+
+### What type of things should regular expressions be used for?
+
+1. Anything that would contain a groupId
+2. Anything that would contain an artifactId
+3. Anything that would contain a version. There are two reasons for this. The first reason because of version tags in pom files. Each version tag should have a version update tag which contains the group and artifact. The second reason is that while our Azure SDK for Java versions will successfully pass spelling checks, it's not a guarantee that external dependency version will. For ex. 2.0-groovy-3.0. Maven accepts these type of version strings.
+4. Anything that contains class names. Classes often contain abbreviations or words that won't pass spellcheck. For example: AMQP.
+5. Anything that contain lists 1-4. For example `<excludePackageNames>` is a multi-line regular expression because it contains a list of packages to exclude.
+
+## Do nots
+
+1. Do not add a name, initials or aliases to the skip list. Individual names, initials or aliases does not belong in anything that ships.
```

Comment:
We do have an exception for this today. We want to validate the spelling that goes into the `[CHANGELOG.md](https://github.dev/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/CHANGELOG.md#L29)` files and sometimes those contain github aliases for contributors. In those cases we [add those aliases to `words`](https://github.dev/Azure/azure-sdk-for-cpp/blob/main/.vscode/cspell.json#L70) ... However, this is a good place to consider adding them to a specific override for `CHANGELOG.md` files so they don't, for example, get scooped up in later efforts to contribute to upstream dictionaries. 

https://github.com/Azure/azure-sdk-for-cpp/issues/3660

Reply:
@danieljurek, Java doesn't put aliases, initials or names into changelogs. I created this set of rules based upon what I saw and, also, what made sense.